src/snowflake/connector/storage_client.py

```diff
@@ -202,6 +205,15 @@ def preprocess(self) -> None:
         if meta.require_compress:
             self.compress()
         self.get_digest()
+
+        if (
+            meta.skip_upload_on_content_match
+            and file_header
+            and meta.sha256_digest == file_header.digest
+        ):
+            logger.debug(f"same file contents for {meta.name}, skipping upload")
+            meta.result_status = ResultStatus.SKIPPED
```

Comment:
I think codecov bot think this line is not covered is because we skip test if `CONNECTION_PARAMETERS_ADMIN` is not found